ci-operator/step-registry/cucushift/hypershift-extended/idp-openldap/cucushift-hypershift-extended-idp-openldap-commands.sh

```diff
@@ -0,0 +1,300 @@
+        volumeMounts:
+        - name: init-ldif-volume
+          mountPath: /tmp
+      volumes:
+      - name: init-ldif-volume
+        secret:
+          secretName: ldap-init-ldif
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ldap
+  namespace: ldap
+spec:
+  selector:
+    app: ldap
+  ports:
+  - protocol: TCP
+    port: 389
+    targetPort: 389
+EOF
+
+    # Wait for LDAP pod to be running and ready
+    echo "Checking for LDAP pod to be running and ready"
+    count=0
+    max_attempts=10
+    while :; do
+        pod_status=$(oc get pods -n ldap --no-headers | grep ldap)
+        if [[ $pod_status == *"1/1"*"Running"* ]]; then
+            echo "LDAP pod is running and ready."
+            break
+        fi
+
+        echo "Waiting for LDAP pod to be running and ready..."
+        sleep 30
+        ((count++))
+
+        if [[ $count -ge $max_attempts ]]; then
+            echo "Timeout waiting for LDAP pod to be running and ready. Exiting."
+            return 1
+        fi
+    done
+}
+
+# Function to configure the identity provider in OpenShift
+function configure_identity_provider() {
+    echo "Configuring OpenShift identity provider with OpenLDAP"
+
+    #Create network policy to allow access from hypershift namespace to ldap namespace
+    cat <<EOF | oc apply -f -
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-ldap-to-hypershift-ci
+  namespace: clusters-$cluster_name
+spec:
+  podSelector: {}
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: ldap
+  egress:
+  - to:
+    - namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: ldap
+EOF
+
+    # current generation
+    gen=$(oc get deployment oauth-openshift -n "clusters-$cluster_name" -o jsonpath='{.metadata.generation}')
+
+    # Prepare the new LDAP identity provider configuration as a JSON object
+    new_ldap_idp='{
+    "spec":{
+        "configuration":{
+            "oauth":{
+                "identityProviders": [{
+                    "ldap": {
+                        "attributes": {
+                            "email": ["mail"],
+                            "id": ["dn"],
+                            "name": ["cn"],
+                            "preferredUsername": ["cn"]
+                        },
+                        "bindDN": "",
+                        "bindPassword": {
+                            "name": ""
+                        },
+                        "insecure": true,
+                        "url": "ldap://ldap.ldap.svc.cluster.local:389/ou=people,ou=rfc2307,dc=example,dc=com?cn"
+                    },
+                    "mappingMethod": "claim",
+                    "name": "ldapidp",
+                    "type": "LDAP"
+                }]
+            }
+    }}}'
+
+    # Apply the patch
+    if ! oc patch hostedclusters $cluster_name -n "$HYPERSHIFT_NAMESPACE" --type=merge -p "$new_ldap_idp"; then
```

Comment:
@dpuniaredhat , submitted https://github.com/openshift/release/pull/53677 . You can refer same to reuse your original openldap ref and script.

Reply:
Ack, Thanks